src/Automation/ScanResultsAssembler.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Axe.Windows.Core.Bases;
+using Axe.Windows.Core.Results;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Axe.Windows.Automation
+{
+    /// <summary>
+    /// Provides methods used to assemble a ScanResults object
+    /// </summary>
+    public static class ScanResultsAssembler
+    {
+        /// <summary>
+        /// Assembles failed scans from the provided element
+        /// </summary>
+        /// <param name="element">Root element from which scan results will be assembled</param>
+        /// <returns>A ScanResults object containing the relevant errors and error count</returns>
+        public static ScanResults AssembleScanResultsFromElement(A11yElement element)
+        {
+            var errors = new List<ScanResult>();
+            int count = AssembleScanResults(errors, element, null);
```

Comment:
This variable isn't really necessary because you can get the count from the returned list with as much clarity.

Reply:
Good point